### PR TITLE
tetrio-desktop: cleanup, fix GApps wrapping, add tetrio-plus option

### PR DESCRIPTION
**Pull request by huantianad**
###### Description of changes

The first commit reorganizes derivation arguments to fit usage order, removes packages from `libPath` which were already patched by `autoPatchelfHook`, and most importantly, properly wraps the program with `wrapGAppsHook`. Previously `dontWrapGApps` was correctly set to prevent double wrapping; however, `gappsWrapperArgs` wasn't passed to `wrapProgram` so no wrapping was done. This properly passes the arguments to `wrapProgram` and moves `wrapProgram` to the fixup phase, as `gappsWrapperArgs` isn't fully populated until then. Proper GTK wrapping is necessary for certain features of `tetrio-plus` to function.

The second commit adds the `tetrio-plus` tool as an optional add-on to `tetrio-desktop`. 

For reviewers:
 - Does removing packages from `libPath` break anything?
 - Is `withTetrioPlus` a good name for the option?
 - Consider building `tetrio-plus` from source. Currently downloads prebuilt version, since it requires manually using `npm i` which doesn't work in a nix sandbox. Maybe it's possible with #189539?

@wackbyte 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wackbyte** [comment]
:+1: Tested with and without `tetrio-plus`. As for the name of the `withTetrioPlus` option, that format seems to have a precedent with Discord's `withOpenASAR` option, so personally I think it's good.

**Comment by huantianad** [comment]
@wackbyte sounds good then! Do we want to wait for additional testing, or to try to get tetrio-plus to build from source before merging?

File: pkgs/games/tetrio-desktop/default.nix [hunk 1]
```diff
@@ -67,6 +69,14 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+  postInstall =
+    let
+      tetrio-plus = callPackage ./tetrio-plus.nix { };
```
Comment (wackbyte) [inline pkgs/games/tetrio-desktop/default.nix:74]: Could you move `tetrio-plus` to the derivation arguments and have it default to this value?

**Comment by wackbyte** [comment]
Since the prebuilt package works fine, building `tetrio-plus` from source can wait until later when it becomes feasible to do so.

File: pkgs/games/tetrio-desktop/default.nix (hunk 1 above)
Comment (huantianad) [reply pkgs/games/tetrio-desktop/default.nix]: Done!

**Comment by wackbyte** [comment]
LGTM!

**Comment by huantianad** [comment]
(I think you should mark the PR as approved via GitHub too, that way OfBorg can add the label)

**Review by wackbyte — APPROVED** [review]

**Comment by huantianad** [comment]
Ok I guess I'm mistaken heh, I'll just add them manually.

**Pull request merged**